model-optimizer/extensions/middle/dequantize_linear_resolver.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from extensions.ops.Cast import Cast
+from extensions.ops.elementwise import Mul, Sub
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, rename_nodes
+from mo.middle.passes.convert_data_type import data_type_str_to_np
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.reshape import Reshape
+
+
+class DequantizeLinearResolver(MiddleReplacementPattern):
+    """
+    Transformation result depend on from axis value
+    If axis not set or default value equal 1 DequantizeLinear can be replace with the following formula:
+        y = (x - x_zero_point) * x_scale
+    In other cases DequantizeLinear can be replace to formula with addition broadcast x_zero_point and x_scale.
+    Target shape for broadcasting depend on axis.
```

Comment:
There is no broadcasting in the transformation. The values `x_zero_point` and `x_scale` are reshaped to a different shape. Explain in the comments how the reshape happens.

Reply:
Changed to reshape